docs/templates.md

```diff
@@ -479,7 +479,9 @@ defined.
 * `.trim_end() -> String`: Removes trailing whitespace
 * `.substr(start: Integer, end: Integer) -> String`: Extract substring. The
   `start`/`end` indices should be specified in UTF-8 bytes. Negative values
-  count from the end of the string.
+  count from the end of the string. Indices are 0-based and `end` is exclusive.
```

Comment:
nit: The "0-based" conflicts a bit with the "negative values" bit before. Maybe something like this:

> Indices are 0-based and `end` is exclusive. Negative values count from the end of the string, with `-1` being the last byte.